packages/frontend/src/model/GatewayNodes/queries/useGateways.ts

```diff
@@ -16,26 +17,34 @@ type Gateway = {
   nodeId: string;
   configTimestamp: number;
   lastPingAt: number;
+  hostname: string;
+  status: string;
 };
 
 const createNodeList = (data: Gateway[]) => {
   if (!data) return [];
 
-  return data.reduce((acc: any[], gateway: Gateway, idx: number) => {
-    // We can mutate acc because we have new [] each time we call this function
-    acc[idx] = [gateway.nodeId, 'active', gateway.lastPingAt, gateway.configTimestamp];
-    return acc;
-  }, []);
+  return data
+    .slice()
+    .sort((a, b) => b.lastPingAt - a.lastPingAt)
+    .reduce((acc: any[], gateway: Gateway, idx: number) => {
+      // We can mutate acc because we have new [] each time we call this function
+      acc[idx] = [
+        gateway.nodeId,
+        gateway.hostname,
+        gateway.status,
+        formatDistanceToNow(gateway.lastPingAt, { addSuffix: true }),
+        gateway.configTimestamp,
+      ];
+      return acc;
+    }, []);
 };
 
 export const useGateways = () => {
-  const { data, loading, error, subscribeToMore } = useQuery(QUERY_GATEWAYS);
-
-  subscribeToMore({
-    document: GATEWAYS_UPDATE,
-    updateQuery: (_, { subscriptionData }) => {
-      return { getGateways: createNodeList(subscriptionData.data.gatewaysUpdated) };
-    },
+  // random interval between 5 and 20 secs
+  const randomPollInterval = (Math.floor(Math.random() * 16) + 5) * 1000;
```

Comment:
what was the problem with the subscription? should we remove it from the backend?

Reply:
There was a problem with authorization and passing an accessToken in the subscription itself. At the moment, the backend doesn't work at all if we are trying to do that. Besides, Apollo themselves do not recommend to use the subscriptions _unless_ the amount of data is important (which is not in our case).